modules/generator/instance.go

```diff
@@ -261,14 +266,27 @@ func (i *instance) pushSpans(ctx context.Context, req *tempopb.PushSpansRequest)
 func (i *instance) updatePushMetrics(req *tempopb.PushSpansRequest) {
 	size := 0
 	spanCount := 0
+	expiredSpans := 0
 	for _, b := range req.Batches {
 		size += b.Size()
 		for _, ils := range b.InstrumentationLibrarySpans {
 			spanCount += len(ils.Spans)
+			// filter spans that have end time > max_age
+			var newSpansArr []*v1.Span
+			timeNow := time.Now().UnixNano()
+			for _, span := range ils.Spans {
+				if span.EndTimeUnixNano >= uint64(timeNow-i.cfg.MaxSpanAge*1000000000) {
```

Comment:
both sides of this time range need to be checked. If the user sends a span that's 5 days in the future it should not impact metrics.

we have similar code in the wal:
https://github.com/grafana/tempo/blob/5c885f37f1a01b07db1be1f6d6ca04ce1a58113c/tempodb/wal/append_block.go#L235

Reply:
Is there a reason why we pick 5 days?

Reply:
I think 5 days was just an example. I think we can start with a symmetrical time range, i.e. use the same duration before and after `time.Now()`. If this doesn't work well we could still break it out into two config options.